### PR TITLE
Support const vector argument for isnan(), atan2()

### LINKED ISSUE #3823: Assertion when calling isnan() with vector
[According to the HLSL documentation](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-isnan) `isnan()` should accept a scalar or a vector:

Name | Template Type | Component Type | Size
-- | -- | -- | --
x | scalar,&nbsp;vector, or&nbsp;matrix | float | any
ret | scalar,&nbsp;vector, or&nbsp;matrix | bool | as input

However attempting to call `isnan()` with a vector produces the following ICE:

```hlsl
void main() {
  bool3 _ = isnan(float3(0.0f, 0.0f, 0.0f));
}
```

```bash
> ./dxc test.hlsl -Tps_6_0
Internal Compiler error: cast<X>() argument of incompatible type!
```

This might be a regression, as the version of DXC used by shader compiler [appears to handle this](http://shader-playground.timjones.io/0800e300a92f9cdf26404549ef54fb0d) (`Updated from trunk on 2020-12-16`).

### PR DESCRIPTION
**Pull request by jaebaek**
Fixes #3823, #3824

### CODE REVIEW

**Comment by jaebaek** [comment]
It is not related to SPIR-V. It is related to the constant folding for the intrinsic function calls with constant arguments. PTAL!

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 1]
```diff
@@ -2877,6 +2877,14 @@ TEST_F(CompilerTest, CodeGenWaveSize) {
...
 }
 
+TEST_F(CompilerTest, CodeGenVectorIsnan) {
+  CodeGenTestCheck(L"isnan_vector_argument.hlsl");
+}
+
+TEST_F(CompilerTest, CodeGenVectorAtan2) {
+  CodeGenTestCheck(L"atan2_vector_argument.hlsl");
+}
+
```
Comment (pow2clk) [inline tools/clang/unittests/HLSL/CompilerTest.cpp:2887]: The TAEF version of this file requires TEST_METHOD invocations in the class above around line 158. That's why the windows builds are failing

**Review by pow2clk — COMMENTED** [review]
This is great! Thanks for doing it!

With the small change to the CompilerTest.cpp file, this should be good to go.

I suppose the advantage of adding these files to CompilerTest instead of just dumping them in the HLSLFileCheck directory is that they will be run on Linux. That's fine for now. I have a 80% completed task to re-enable the filecheck test on Linux, but I can't justify blocking this behind that.

File: tools/clang/unittests/HLSL/CompilerTest.cpp (hunk 1 above)
Comment (jaebaek) [reply tools/clang/unittests/HLSL/CompilerTest.cpp:2889]: done

**Review by pow2clk — APPROVED** [review]
Looks good!

Thanks again for doing this!

**Comment by jaebaek** [comment]
My pleasure!

**Pull request merged**